python/lsst/analysis/tools/actions/plot/matrixPlot.py

```diff
@@ -0,0 +1,538 @@
+        ):
+            raise ValueError(
+                "If 'component1Key' and 'component2Key' are provided, 'xAxisTickValues', "
+                "'yAxisTickValues', 'xAxisTickLabels', and 'yAxisTickLabels' should not be "
+                "provided as they will be dynamically configured."
+            )
+
+    def makePlot(self, data: KeyedData, plotInfo: Mapping[str, str] | None = None, **kwargs: Any) -> Figure:
+        """
+        Plot a matrix of values.
+
+        Parameters
+        ----------
+        data : `~lsst.analysis.tools.interfaces.KeyedData`
+            The data to plot.
+        plotInfo : `dict`, optional
+            A dictionary of information about the data being plotted.
+        **kwargs
+            Additional keyword arguments to pass to the plot.
+
+        Returns
+        -------
+        fig : `~matplotlib.figure.Figure`
+            The resulting figure.
+        """
+        # Retrieve the matrix info from the input data.
+        matrix = data[self.matrixKey]
+
+        # Fetch the components between which the correlation is calculated.
+        if self.component1Key is not None and self.component2Key is not None:
+            comp1 = data[self.component1Key]
+            comp2 = data[self.component2Key]
+
+        if self.inputDim == 1:
+            # Calculate the size of the square.
+            square_size = int(np.sqrt(matrix.size))
+            # Reshape into a square array.
+            matrix = matrix.reshape(square_size, square_size)
+            if self.component1Key is not None and self.component2Key is not None:
+                comp1 = comp1.reshape(square_size, square_size)
+                comp2 = comp2.reshape(square_size, square_size)
+
+        # Calculate default limits only if needed.
+        if self.vmin is None or self.vmax is None:
+            default_limits = apViz.PercentileInterval(98.0).get_limits(np.abs(matrix.flatten()))
+        else:
+            default_limits = (None, None)
+
+        # Set the value range using overrides or defaults.
+        vrange = (
+            default_limits[0] if self.vmin is None else self.vmin,
+            default_limits[1] if self.vmax is None else self.vmax,
+        )
+
+        # Allow for the figure object to be passed in.
+        fig = kwargs.get("fig")
+        if fig is None:
+            fig = plt.figure(figsize=self.figsize, dpi=self.dpi)
+            ax = fig.add_subplot(111)
+        else:
+            ax = fig.gca()
+
+        if self.title:
+            ax.set_title(self.title, fontsize=self.titleFontSize)
+
+        if self.xAxisLabel:
+            ax.set_xlabel(self.xAxisLabel, fontsize=self.axisLabelFontSize)
+
+        if self.yAxisLabel:
+            ax.set_ylabel(self.yAxisLabel, fontsize=self.axisLabelFontSize)
+
+        # Set the colorbar and draw the image.
+        norm = ImageNormalize(vmin=vrange[0], vmax=vrange[1])
+        img = ax.imshow(matrix, interpolation="none", norm=norm)
```

Comment:
I think the `origin` kwarg should be configurable here (I prefer `lower` which is unambiguously the correct choice for our actual CCD images but making it a `ChoiceField[str]` with `upper` as the default is fine if that's the convention for covariance plots for the camera team).

Reply:
Done! I have made it configurable so that we can revisit this decision in the future.